cowprotocol/cowswap
# fix(twap): display twap orders history

# Summary

Fixes #3549

The source of the bug: https://github.com/cowprotocol/cowswap/pull/3416/files#diff-a3fa475dfa20444e1700ef4f426bfadd6f82c28c834bc612d7bd981b9440ffd0R33

So, while refactoring multicalls I also changed `useTwapOrdersAuthMulticall` hook and introduced the bug there.
Before the refactoring `useSingleContractMultipleData` used to return empty array when there are no input requests.
After the refactoring it started returning `null` which is supposed to be useful in case if you want to distinguish cases when multicall result empty and when multicall even didn't happen.

`TwapOrdersUpdater`checks `if (!ordersAuthResult) return` to prevent orders updating before we get authentication results. But in case when we don't have any pending orders we don't check authentication.

  # To Test

1. Open https://app.safe.global/apps/open?safe=gno:0x42cEDde51198D1773590311E2A340DC06B24cB37&appUrl=https://swap-dev-git-fix-twap-orders-history-cowswap.vercel.app
2. Check TWAP and Limit orders history tabs in the Your orders page
- [ ] AR: no TWAP orders history --> orders are displayed in the Limit orders history
- [ ] ER: separate Limit orders and TWAP orders
